### DISCUSSION TITLE
开启透明代理启动节点报错，不开启可以正常启动节点

### DISCUSSION THREAD

**Discussion by KPdata** [v2rayA/v2rayA#1437] [Q&A]
> 使用系统：Alpine Linux v3.20.1

各位大佬，我这边根据Alpine系统下的部署教程[https://v2raya.org/docs/prologue/installation/alpine/](https://v2raya.org/docs/prologue/installation/alpine/)安装好了V2Ray内核和v2rayA，在不开启**透明代理**的设置下可以正常启动节点，但一开启就报错。不知有没有大佬碰到过这种问题😭

**报错1：**
 `failed to start v2ray-core: not support "redirect" mode of transparent proxy: ExecCommands: iptables-legacy -w 2 -t nat -N TP_OUT sh: iptables-legacy: not found : exit status 127`
我看报错提示`iptables-legacy: not found`随后我直接进行一个`apk add iptables-legacy`，随后再次尝试启动透明代理，结果又来了一个新的报错。
**报错2：**
`failed to start v2ray-core: not support "redirect" mode of transparent proxy: ExecCommands: iptables-legacy -w 2 -t nat -N TP_OUT iptables v1.8.10 (legacy): can't initialize iptables tablenat': Permission denied (you must be root) Perhaps iptables or your kernel needs to be upgraded. : exit status 3`

**Comment by tobybo** [comment]
我也是，我手动安装升级了 Iptables ，用docker部署的。
![Uploading image.png…]()

**Reply by lukea99** [reply]
怎样手动升级iptables？

**Reply by F7kyyy** [reply]
> 怎样手动升级iptables？

解决了么，我现在firewalld 依赖的iptables是1.8.7也报这个错，升级到什么版本？

**Comment by taurusleo** [comment]
换一个mode就可以了

**Comment by MR-WANGG** [comment]
大佬，我这个也开启不了，该如何解决呀？提示如下：

failed to start v2ray-core: not support "redirect" mode of transparent proxy: ExecCommands: iptables-legacy -w 2 -t nat -N TP_OUT iptables v1.8.10 (nf_tables): Could not fetch rule set generation id: Permission denied (you must be root) : exit status 4

**Reply by MR-WANGG** [reply]
求助大佬，折腾的头晕眼花，一团浆糊了，不知道怎么弄了。

failed to start v2ray-core: not support "redirect" mode of transparent proxy: ExecCommands: iptables-legacy -w 2 -t nat -N TP_OUT iptables v1.8.10 (nf_tables): Could not fetch rule set generation id: Invalid argument : exit status 4

**Reply by MR-WANGG** [reply]
最高root权限运行的

**Reply by li-bri** [reply]
+1，遇到了同样的问题，我是在arm64的ubuntu平台上运行的，请问有影响吗？

**Reply by amenzhj** [reply]
https://github.com/v2rayA/v2rayA/discussions/1600#discussioncomment-12555190

**Comment by HUYANG1994** [comment]
我也是 你们怎么决绝的

**Comment by laopaoer-wallet** [comment]
开启不了透明代理 docker 部署

**Reply by shadowzym** [reply]
运行 iptables -V 检查系统使用的是 iptables ( nf_tables ) 还是 iptables ( legacy )。如果是nf_tables，需要设置环境变量V2RAYA_NFTABLES_SUPPORT=on ；如果是legacy，添加环境变量IPTABLES_MODE=legacy ，此外可能还要检查并加载 xt_TPROXY 内核模块。编辑补充：关于Permission denied问题，看下--cap-add给的权限是否足够，否则用--privileged
